browser/ui/brave_shields_data_controller_unittest.cc

```diff
@@ -0,0 +1,187 @@
+
+    web_contents_ =
+        content::WebContentsTester::CreateTestWebContents(profile_, nullptr);
+    BraveShieldsDataController::CreateForWebContents(web_contents_.get());
+  }
+
+  void TearDown() override {
+    web_contents_.reset();
+    profile_ = nullptr;
+    profile_manager_->DeleteTestingProfile(kTestProfileName);
+  }
+
+  Profile* profile() { return profile_; }
+  content::WebContents* web_contents() { return web_contents_.get(); }
+
+  void SetLastCommittedUrl(const GURL& url) {
+    content::WebContentsTester::For(web_contents())->SetLastCommittedURL(url);
+  }
+
+  BraveShieldsDataController* GetShieldsDataController() {
+    return BraveShieldsDataController::FromWebContents(web_contents());
+  }
+
+  ContentSetting GetContentSettingFor(ContentSettingsType type,
+                                      GURL secondary_url = GURL()) {
+    auto* map = HostContentSettingsMapFactory::GetForProfile(profile());
+    auto* controller = GetShieldsDataController();
+
+    return map->GetContentSetting(controller->GetCurrentSiteURL(),
+                                  secondary_url, type);
+  }
+
+  ContentSettingsPattern GetPatternFromURL(const GURL& url) {
+    DCHECK(url.is_empty() ? url.possibly_invalid_spec() == "" : url.is_valid());
+    if (url.is_empty() && url.possibly_invalid_spec() == "")
+      return ContentSettingsPattern::Wildcard();
+    return ContentSettingsPattern::FromString("*://" + url.host() + "/*");
+  }
+
+  void SetContentSettingFor(ContentSettingsType type,
+                            ContentSetting setting,
+                            GURL secondary_url = GURL()) {
+    auto* map = HostContentSettingsMapFactory::GetForProfile(profile());
+    auto* controller = GetShieldsDataController();
+
+    map->SetContentSettingCustomScope(
+        GetPatternFromURL(controller->GetCurrentSiteURL()),
+        GetPatternFromURL(secondary_url), type, setting);
+  }
+
+ private:
+  content::BrowserTaskEnvironment task_environment_;
+  std::unique_ptr<content::WebContents> web_contents_;
+  Profile* profile_;
+  std::unique_ptr<TestingProfileManager> profile_manager_;
+
+  DISALLOW_COPY_AND_ASSIGN(BraveShieldsDataControllerTest);
+};
+
+TEST_F(BraveShieldsDataControllerTest, SetAdBlockMode_ForOrigin) {
+  auto* controller = GetShieldsDataController();
+  SetLastCommittedUrl(GURL("http://brave.com"));
+
+  /* DEFAULT */
+  EXPECT_EQ(GetContentSettingFor(ContentSettingsType::BRAVE_ADS),
+            CONTENT_SETTING_DEFAULT);
+  EXPECT_EQ(GetContentSettingFor(ContentSettingsType::BRAVE_TRACKERS),
+            CONTENT_SETTING_DEFAULT);
+  EXPECT_EQ(GetContentSettingFor(ContentSettingsType::BRAVE_COSMETIC_FILTERING),
+            CONTENT_SETTING_DEFAULT);
+  EXPECT_EQ(GetContentSettingFor(ContentSettingsType::BRAVE_COSMETIC_FILTERING,
+                                 GURL("https://firstParty/")),
+            CONTENT_SETTING_DEFAULT);
+
+  /* SET BLOCK MODE TO ALLOW */
+  controller->SetAdBlockMode(AdBlockMode::ALLOW);
```

Comment:
I think we probably want to split these into multiple test cases so we can test going back and forth 
default -> allow -> standard -> allow -> block -> allow
default -> standard -> allow -> standard -> block -> standard
default -> block -> allow -> block -> standard -> block
I think that covers every possible transition

Reply:
done

Reply:
this change was for `Set`, sorry for the misunderstanding